contrib/pg_upgrade/check.c

```
@@ -342,13 +342,11 @@ check_cluster_versions(void)
 	 */
 
 	/*
-	 * Upgrading from anything older than an 8.2 based Greenplum is not
-	 * supported. TODO: This needs to be amended to check for the actual
-	 * 4.3.x version we target and not a blanket 8.2 check, but for now
-	 * this will cover most cases.
```

Why this? `pg_upgrade` should not be limited to upgrading from just the previous version, and we still support 4.3.x upgrades in the code so this check seems wrong now?

---

Yeah, I'm confused about this too.

---

I thought to tighten up the check to only allow 5 and later to be upgraded since it's likely the 4-6 upgrade won't "just work".  But I can see the argument to leave it and let it fail if and where it does during an actual upgrade.

---

4-6 should just work, thats the main idea behind pg_upgrade, being able to upgrade any previous version to the current.

Now, I know too that we're not really done with this tool and that there are bugs, but we should stick to the overall plan still.

---

I reverted this change.